libraries/botbuilder-dialogs-adaptive/src/recognizers/recognizerSet.ts

```diff
@@ -63,13 +64,15 @@ export class RecognizerSet extends Recognizer implements RecognizerSetConfigurat
 
                 // merge intents
                 for (const intentName in result.intents) {
```

Comment:
Since you're here - can we rewrite this to use `for...of` instead of `for...in`? It's generally a more sane option.

Reply:
An example might be:

```typescript
for (const [intentName, intent] of Object.entries(result.intents)) {
  ...
}
```

Reply:
Refactored to use for-of loops!

Although separately, going through this recognizer for the refactor, it merges the `RecognizerResult`'s of multiple recognizers in a *really* bizarre manner:
`RecognizerResult`
- Text: whatever is the last `RecognizerResult` in the array of results from multiple recognizers registered, will "win" the spot of Text in the merged `RecognizerResult`
- Intent: In C# the method doc includes this "`Intents will be merged by picking the intent with the MaxScore.`". However they both actually just push all intents recognized from each `RecognizerResult` in the array of results into the merged `RecognizerResult`
- Entities: Contrast with the doc in C# regarding entities "`Entities are merged as a simple union of all of the Entities.`" -- this is the actual behavior what's happening (all entities recognized by all recognizers are included in the merged `RecognizerResult`)

The behavior is the same in both C# and JS, however it seems buggy in both, if we're going off by what the C# doc says vs. what is actually happening...iunno. I don't know enough about the design decision around `RecognizerSet` to make any behavioral changes